pkg/assembler/helpers/purl.go

```diff
@@ -42,6 +42,30 @@ func PurlToPkg(purlUri string) (*model.PkgInputSpec, error) {
 	return purlConvert(p)
 }
 
+func AllPkgTreeToPurl(v *model.AllPkgTree) string {
+	ns := v.Namespaces[0]
```

Comment:
There can be more than one actual software package in this trie, but this function only extracts one. To cover the general case I think that iteration over the namespaces, names, and versions is needed.

Reply:
Based on the query used, it could only return one or many. What is the context around this @lumjjb?

Reply:
yea this is for singular use, let me add a docstring to explain the utility.

Reply:
So it will always be a singular result?

Reply:
i was doing this for https://github.com/guacsec/guac/blob/recursive-known2/cmd/guacone/cmd/known.go#L497-L515, so am happy to just roll this into a private function but thought it'd be useful

Reply:
with the comment there it should be good.